common/http.go

```diff
@@ -31,6 +31,7 @@ type HTTPClient struct {
 	Base         string                  // optional base URL. If not set, then all URLs must be absolute.
 	Client       AuthenticatedHTTPClient // underlying HTTP client. Required.
 	ErrorHandler ErrorHandler            // optional error handler. If not set, then the default error handler is used.
+	OKStatusErr  bool                    // optional, If set true indicates 200OK http response may include errors.
```

Comment:
I see why you added this, but how about having a `ResponseHandler` instead? We can add that in place of this. You would shift the response handling in `sendRequest` to this `ResponseHandler` - i.e. 
```
type ResponseHandler func(rsp *http.Response) (*http.Response, error)

// sendRequest sends the given request and returns the response & response body.
func (h *HTTPClient) sendRequest(req *http.Request) (*http.Response, error) { //nolint:cyclop
	// Send the request
	res, err := h.Client.Do(req)
	if err != nil {
		return nil, err
	}

	// Apply the ResponseHandler if provided
	if h.ResponseHandler != nil {
		res, err = h.ResponseHandler(res)
                 ......
```

Since it's optional, if there's no ResponseHandler, there will be no change. But for Marketo, you can set the ResponseHandler to parse it and convert it into a http Response with the error code in the body set to the http.Response.Code